TutorLizard.BusinessLogic/Services/TutorService.cs

```diff
@@ -14,57 +14,44 @@ public TutorService(ITutorDataAccess dataAccess, IUserIdentityService userIdenti
     }
     public Ad CreateAd(string subject, string title, string description)
     {
-        // return created Ad
-        // return empty with id = 0 to indicate something went wrong
-        return new Ad()
+        int? tutorId = _userIdentityService.GetUserId();
+        if (tutorId is null)
         {
-            // this is only for tests
-            Id = 1,
-            Subject = subject,
-            Title = title,
-            Description = description
-        };
+            return new Ad()
+            { Id = 0 };
+        }
+
+        return _dataAccess.CreateAd((int)tutorId, subject, title, description);
+
     }
     public ScheduleItem CreateScheduleItem(int adId, DateTime dateTime)
     {
-        // return created ScheduleItem
-        // return empty with id = 0 to indicate something went wrong
-        return new ScheduleItem()
-        {
-            // this is only for tests
-            Id = 1,
-            AdId = adId,
-            DateTime = dateTime
-        };
+        return _dataAccess.CreateScheduleItem(adId, dateTime);
     }
-    public Ad? GetAdById(int adId)
+    public Ad? GetAdById(int? adId)
     {
         // return Ad (from _dataAccess) with provided adId
         // return null if no such Ad
-        return new Ad()
+        adId = _dataAccess.GetAllAdsId();
+        if (adId is null)
         {
-            // this is only for tests
-            Id = adId,
-            Title = "Test"
-        };
+            return null;
+        }
+        return _dataAccess.GetAdById((int)adId);
     }
-    public ScheduleItem? GetScheduleItemById(int scheduleItemId)
+    public ScheduleItem? GetScheduleItemById(int? scheduleItemId)
     {
-        return new ScheduleItem()
+        scheduleItemId = _dataAccess.GetScheduleItemId();
+        if (scheduleItemId is null)
         {
-            // this is only for tests
-            Id = scheduleItemId,
-            DateTime = DateTime.Now.AddHours(1),
-        };
+            return null;
+        }
+        return _dataAccess.GetScheduleItemById(scheduleItemId);
     }
     public List<Ad> GetUsersAds()
     {
-        return [
-            // this is only for tests
-            GetAdById(1)!,
-            GetAdById(2)!,
-            GetAdById(3)!
-            ];
+        List<Ad> ads = new List<Ad>();
+        return _dataAccess.GetUsersAds().ToList();
```

Comment:
If you want to move filtering to `DataAccess`, you need to at least pass user's Id as a parameter. Otherwise, it has no way of knowing what you want.
I wanted to say that I think it would be cleaner to get all ads and filter them yourself with `Where()`, but on second thought, if we changed our `DataAccess` to an SQL database, it would actually be **much** better to tell `DataAccess` to get you exactly what you want, so please leave it like this, but add the parameter.